test/jdk/java/lang/module/ModuleReader/patched/PatchedModuleReaderTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.IOException;
+import java.lang.module.ModuleFinder;
+import java.lang.module.ModuleReader;
+import java.lang.module.ModuleReference;
+import java.net.URI;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/*
+ * @test
+ * @bug 8372787
+ * @summary verify the behaviour of ModuleReader when using --patch-module
+ * @comment patch the java.base module with a test specific resource
+ * @compile/module=java.base java/lang/PatchedFoo.java
+ * @run junit/othervm ${test.main.class}
+ */
+class PatchedModuleReaderTest {
+
+    private static ModuleReference patchedModuleRef;
+
+    @BeforeAll
+    static void beforeAll() {
+        patchedModuleRef = ModuleFinder.ofSystem()
+                .find("java.base")
+                .orElseThrow();
+    }
+
+    /*
+     * Verifies that the resource that was patched into a module
+     * is found by the ModuleReader.
+     */
+    @Test
+    void testResourceFound() throws Exception {
+        try (ModuleReader reader = patchedModuleRef.open()) {
+            String resourceName = "java/lang/PatchedFoo.class";
+            Optional<URI> res = reader.find(resourceName);
+            assertTrue(res.isPresent(), resourceName + " is missing in "
+                    + patchedModuleRef.descriptor().name() + " module");
```

Comment:
A suggestion here is replace this assert with a more complete test to check that the URI's scheme is "file" and that its path component ends with "java/lang/PatchedFoo.class".